ci/run_envoy_docker.sh

```diff
@@ -91,6 +91,7 @@ docker run --rm \
        "${VOLUMES[@]}" \
        -e AZP_BRANCH \
        -e AZP_COMMIT_SHA \
+       -e AZP_TARGET_BRANCH \
```

Comment:
i have added this separate to `AZP_BRANCH` as it doesnt work quite the same way so it would have been ambiguous to just use that